### PR TITLE
Refine tweak changed notification logic. Simplify MutableTweaksConfiguration.

### PR DESCRIPTION
**Pull request by albertodebortoli**
Follow-up on #29 cc @federicocappelli

- [x] Refine Tweak changed notification logic
- [x] Simplify MutableTweaksConfiguration protocol

### CODE REVIEW

File: JustTweak/Classes/Core/TweaksConfigurationsCoordinator.swift [hunk 1]
```diff
@@ -127,14 +127,14 @@ final public class TweaksConfigurationsCoordinator: NSObject, TweaksConfiguratio
...
+    public func registerForConfigurationsUpdates(_ object: NSObject, closure: @escaping (Tweak) -> Void) {
         deregisterFromConfigurationsUpdates(object)
         let queue = OperationQueue.main
         let name = TweaksConfigurationDidChangeNotification
         let notificationsCenter = NotificationCenter.default
-        let observer = notificationsCenter.addObserver(forName: name, object: nil, queue: queue) { (notification) in
-            let tweakIdentifier = notification.userInfo?[TweaksConfigurationDidChangeNotificationTweakIdentifierKey] as? String
-            closure(tweakIdentifier)
+        let observer = notificationsCenter.addObserver(forName: name, object: nil, queue: queue) { notification in
+            guard let tweak = notification.userInfo?[TweaksConfigurationDidChangeNotificationTweakKey] as? Tweak else { return }
```
Comment (albertodebortoli) [inline JustTweak/Classes/Core/TweaksConfigurationsCoordinator.swift]: This observer is listening to a notification that always contains the `Tweak` object. No need to have it optional.

File: Example/Tests/Core/TweaksConfigurationCoordinatorTests.swift [hunk 2]
```diff
@@ -67,20 +67,22 @@ class TweaksConfigurationCoordinatorTests: XCTestCase {
         var didCallClosure = false
         configurationCoordinator.registerForConfigurationsUpdates(self) { tweakIdentifier in
             didCallClosure = true
-            print("\(tweakIdentifier ?? "Unknown") tweak changed")
```
Comment (albertodebortoli) [inline Example/Tests/Core/TweaksConfigurationCoordinatorTests.swift]: No print statements in tests.

File: JustTweak/Classes/Core/TweaksConfiguration.swift [hunk 3]
```diff
@@ -16,26 +16,8 @@ public protocol TweaksConfiguration {
...
-            self.set(value as! Bool, feature: feature, variable: variable)
-        case is String:
-            self.set(value as! String, feature: feature, variable: variable)
-        default:
-            if let value = NSNumber(tweakValue: value) {
-                self.set(value, feature: feature, variable: variable)
-            }
-        }
-    }
+    func set(_ value: TweakValue, feature: String, variable: String)
```
Comment (albertodebortoli) [inline JustTweak/Classes/Core/TweaksConfiguration.swift]: I wanted to simplify this protocol so badly for too long.

File: JustTweak/Classes/Core/TweaksConfigurationsCoordinator.swift (hunk 1 above)
Comment (federicocappelli) [reply JustTweak/Classes/Core/TweaksConfigurationsCoordinator.swift]: *Protective Mode ON* True, but if somebody else, for any reason send you the same notification without a Tweak object you crash the app *Protective Mode OFF*

**Comment by federicocappelli** [comment]
All good from my point of view, thank you 👍

File: Example/JustTweak/ViewController.swift [hunk 4]
```diff
@@ -39,8 +39,8 @@ class ViewController: UIViewController {
     override func viewDidLoad() {
         super.viewDidLoad()
         updateView()
-        configurationsCoordinator?.registerForConfigurationsUpdates(self, closure: { [weak self] (tweakIdentifier) in
-            print("\(tweakIdentifier ?? "Unknown") tweak changed")
+        configurationsCoordinator?.registerForConfigurationsUpdates(self, closure: { [weak self] tweak in
```
Comment (albertodebortoli) [inline Example/JustTweak/ViewController.swift]: can use trailing closure

File: JustTweak/Classes/Core/UserDefaultsTweaksConfiguration.swift [hunk 5]
```diff
@@ -41,23 +41,16 @@ final public class UserDefaultsTweaksConfiguration: NSObject, MutableTweaksConfi
...
-        updateUserDefaultsWith(value: value, feature: feature, variable: variable)
-    }
-    
-    public func set(_ value: NSNumber, feature: String, variable: String) {
-        updateUserDefaultsWith(value: value, feature: feature, variable: variable)
-    }
-    
-    private func updateUserDefaultsWith(value: Any, feature: String, variable: String) {
+        
+    private func updateUserDefaultsWith(value: TweakValue, feature: String, variable: String) {
```
Comment (albertodebortoli) [inline JustTweak/Classes/Core/UserDefaultsTweaksConfiguration.swift]: better to be
```
func updateUserDefaults(with value: TweakValue, feature: String, variable: String)
```

**Comment by dchakarov** [comment]
LGTM

File: JustTweak/Classes/Core/TweaksConfigurationsCoordinator.swift (hunk 1 above)
Comment (albertodebortoli) [reply JustTweak/Classes/Core/TweaksConfigurationsCoordinator.swift]: @federicocappelli it's guarding on that key, it won't crash the app

**Pull request merged**